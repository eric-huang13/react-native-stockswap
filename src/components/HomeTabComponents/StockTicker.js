import React, {PureComponent} from 'react';
import {StyleSheet, Text, View, SafeAreaView} from 'react-native';
import TextTicker from 'react-native-text-ticker';
import {connect} from 'react-redux';
import {moderateScale} from '../../util/responsiveFont';
import {
  fetchTickers,
  fetchMarketGainers,
  fetchMarketLosers,
} from '../../actions/marketMovers';

export class StockTicker extends PureComponent {
  componentDidMount() {
    this.props.fetchGainers();
    this.props.fetchLosers();
  }
  render() {
    const {marketGainers, marketLosers} = this.props;

    const shuffledTickers = marketGainers.flatMap((e, idx) => [
      e,
      marketLosers[idx],
    ]);
console.log(shuffledTickers[shuffledTickers.length-1],"TICKERS")
    return (
      <View style={styles.container}>
        {shuffledTickers[shuffledTickers.length-1] ? (
          <TextTicker
            // style={{fontSize: 18}}
            scrollSpeed={5000}
            duration={shuffledTickers.length * 1000}
            loop
            bounce
            repeatSpacer={0}
            marqueeDelay={0}
            animationType="scroll">
            {shuffledTickers.map((item) => (
              item.ticker !== undefined ?
              <React.Fragment key={item.ticker}>
                <Text
                  style={
                    Math.sign(item.change) < 0
                      ? {...styles.marqueeSymbol, color: '#F66E6E'}
                      : {...styles.marqueeSymbol}
                  }>
                  {' '}
                  {item.ticker}{' '}
                </Text>
                <Text
                  style={
                    Math.sign(item.change) < 0
                      ? {...styles.marqueePercentage, color: '#F66E6E'}
                      : {...styles.marqueePercentage}
                  }>
                  {item.change}%{'   '}
                </Text>
              </React.Fragment>
              : null
            ))}
          </TextTicker>
        ) : null}
      </View>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    marketGainers: state.company.marketGainers,
    marketLosers: state.company.marketLosers,
    // tickers: state.company.tickers,
  };
};
const mapDispatchToProps = (dispatch) => {
  return {
    // fetchTickers: () => dispatch(fetchTickers()),
    fetchGainers: () => dispatch(fetchMarketGainers()),
    fetchLosers: () => dispatch(fetchMarketLosers()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(StockTicker);

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#2e3955',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: moderateScale(6),
  },
  textstyle: {},
  marqueeSymbol: {
    color: '#91f2b1',
    fontSize: moderateScale(14),
    fontFamily: 'Montserrat-Bold',
  },
  marqueePercentage: {
    color: '#91f2b1',
    fontSize: moderateScale(13),
    fontFamily: 'Montserrat-Medium',
  },
});
