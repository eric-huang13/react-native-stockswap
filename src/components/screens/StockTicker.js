import React, {PureComponent} from 'react';
import {StyleSheet, Text, SafeAreaView} from 'react-native';
import TextTicker from 'react-native-text-ticker';
import {connect} from 'react-redux';

export class StockTicker extends PureComponent {
  render() {
    const {gainers} = this.props;

    return (
      <SafeAreaView style={styles.container}>
        <TextTicker
          style={{fontSize: 24}}
          scrollSpeed={5000}
          loop
          bounce
          repeatSpacer={0}
          marqueeDelay={0}
          animationType="scroll">
          {gainers.map((item) => {
            return (
              <>
                <Text
                  style={
                    item.percentage[0] === '-'
                      ? {...styles.marqueeSymbol, color: 'red'}
                      : {...styles.marqueeSymbol}
                  }>
                  {' '}
                  {item.symbol}{' '}
                </Text>
                <Text
                  style={
                    item.percentage[0] === '-'
                      ? {...styles.marqueePercentage, color: 'red'}
                      : {...styles.marqueePercentage}
                  }>
                  {item.percentage}
                  {'  '}
                </Text>
              </>
            );
          })}
        </TextTicker>
      </SafeAreaView>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    gainers: state.company.gainers,
  };
};

export default connect(mapStateToProps)(StockTicker);

const styles = StyleSheet.create({
  container: {
    // flex: 1,
    backgroundColor: '#2e3955',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 5,
  },
  textstyle: {
    color: 'yellow',
  },
  marqueeSymbol: {
    color: '#91f2b1',
    fontSize: 16.5,
    fontWeight: 'bold',
  },
  marqueePercentage: {
    color: '#91f2b1',
    fontSize: 14,
  },
});
