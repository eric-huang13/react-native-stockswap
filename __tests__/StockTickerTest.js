import React from 'react';
import StockTicker from '../src/components/HomeTabComponents/StockTicker';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6

jest.mock('@react-native-community/google-signin', () => {});

jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<StockTicker gainers={[]} />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });