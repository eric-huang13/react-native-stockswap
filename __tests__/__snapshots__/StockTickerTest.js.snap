// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2e3955",
      "justifyContent": "center",
      "marginBottom": 8.474452554744525,
    }
  }
>
  <TextMarquee
    animationType="scroll"
    bounce={true}
    bounceDelay={0}
    bounceSpeed={50}
    disabled={false}
    easing={[Function]}
    isInteraction={true}
    loop={true}
    marqueeDelay={0}
    marqueeOnMount={true}
    repeatSpacer={0}
    scroll={true}
    scrollSpeed={5000}
    shouldAnimateTreshold={0}
    style={
      Object {
        "fontSize": 24,
      }
    }
    useNativeDriver={true}
  />
</ForwardRef(SafeAreaView)>
`;
